core/src/main/java/org/apache/iceberg/ManifestEntriesTable.java

```diff
@@ -69,18 +69,35 @@ MetadataTableType metadataTableType() {
 
   private static class EntriesTableScan extends BaseTableScan {
 
-    EntriesTableScan(TableOperations ops, Table table, Schema schema) {
+    private final String scannedTableName;
+
+    EntriesTableScan(TableOperations ops, Table table, Schema schema, String scannedTableName) {
       super(ops, table, schema);
+      this.scannedTableName = scannedTableName;
     }
 
-    private EntriesTableScan(TableOperations ops, Table table, Schema schema, TableScanContext context) {
+    private EntriesTableScan(TableOperations ops, Table table, Schema schema, String scannedTableName,
+                             TableScanContext context) {
       super(ops, table, schema, context);
+      this.scannedTableName = scannedTableName;
+    }
+
+    @Override
+    public TableScan appendsBetween(long fromSnapshotId, long toSnapshotId) {
+      throw new UnsupportedOperationException(
+          String.format("Incremental scan is not supported for metadata table %s", scannedTableName));
+    }
+
+    @Override
+    public TableScan appendsAfter(long fromSnapshotId) {
+      throw new UnsupportedOperationException(
+          String.format("Incremental scan is not supported for metadata table %s", scannedTableName));
```

Comment:
Could we switch the constructor change to a abstract method? Then every subclass could just add in their own table type something like?

String.format("Incremental scan is not supported for $tableType scan of table $table.name")

Where tableType is defined by subclasses and table is just the table

Reply:
Yeah. We can do that. But `$table.name` will also need to be another abstract method. As the `table` object that is passed in doesn't reference the `#history` table or the specific metadata table (for the use cases where it's a metadata table).

Reply:
Why do we need that when the subclass knows what kind of metadata table it is?

Reply:
I guess we don't, especially if we want it to say `Incremental scan is not supported for HISTORY scan of table $table.name", where "$table.name" would be "foo" and not "foo#history".

I would be ok with that.

I think I misunderstood your original request for the change in the wording (it will no longer say "table foo#history", but the "for HISTORY scan of table foo" will suffice) 👍

Reply:
Yep that's what I was suggesting. If we really wanted it you could just do table

History scan of table $tableName.$metadataIdentifier (which would only be wrong for hadoop tables)

Reply:
So that makes sense.

I got it to work with `Incremental scan is not supported for %s scan of table %s` with the first one being all possible values in `MetadataTableType.values()` and the table always being `table().name()` (e.g. without the $metadataIdentifier).

However, not having the actual scanned table name still leaves us without a solution to this issue, that the events emitted reference the table itself and are potentially incorrect. https://github.com/apache/iceberg/issues/2635

Maybe we can meet Monday or Tuesday and discuss this further? Its possible https://github.com/apache/iceberg/issues/2635 is not a bug. Maybe you can help me better understand the use case for the events, such as `ScanEvent`?

It would seem to me that if we emit a `ScanEvent` when doing an scan on one of the AllMetadataTables, we'd want the emitted scan event to use the scanned table name (e.g. $tableName[.|#]$metadataIdentifier), to distinguish from a scan event on the table's data itself. But possibly that's not necessary?

I can push the updated code and then roll back if we decide not to go with it after we talk.

Reply:
Actually, I think that this is fine.

I'm not too sure I fully understand the use case of `ScanEvent`, so I'm not going to worry about whether or not it's a bug at this moment in time.

I agree that it's more consistent with the rest of the code to be using a method that's overridden vs a constructor parameter.

I have updated to do that, and for the few classes inherit directly from `BaseTableScan`, I've used the same named helper function to keep things somewhat consistent.